### PR TITLE
Creator vitest

### PR DESCRIPTION
**Pull request by NishantSinghhhhh**
What kind of change does this PR introduce?
Added tests for creator.ts.
Issue Number: Fixes: #3064 

Snapshots/Videos:

Summary
This PR adds test cases for creator.ts to ensure better reliability and maintainability. The tests cover key functionalities and edge cases, improving the overall test coverage of the project.

Does this PR introduce a breaking change?
No.

Checklist
CodeRabbit AI Review
 I have reviewed and addressed all critical issues flagged by CodeRabbit AI.
 I have implemented or provided justification for each non-critical suggestion.
 I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented.
Test Coverage
 I have written tests for all new changes/features.
 I have verified that test coverage meets or exceeds 95%.
 I have run the test suite locally, and all tests pass.
Other information

Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?
Yes.

## Release Notes

- **New Features**
  - Added support for specifying natural language preferences during user registration and profile updates
  - Introduced standardized language and country code enumerations

- **Documentation**
  - Updated GraphQL schema to include comprehensive language and country code references

- **Chores**
  - Enhanced type safety for configuration and migration files
  - Expanded database schema with multiple new tables and relationships

- **Testing**
  - Added comprehensive test coverage for new advertisement creator resolver and user language preference features

### CODE REVIEW

**Comment by NishantSinghhhhh** [comment]
@palisadoes , I have raised a PR for adding test to creator.ts , what more should I do so that i can make this PR mergeable

File: drizzle_migrations/20250130163044_mature_james_howlett.sql [hunk 1]
```diff
@@ -0,0 +1,675 @@
+CREATE TABLE "action_categories" (
+	"created_at" timestamp (3) with time zone DEFAULT now() NOT NULL,
+	"creator_id" uuid,
+	"description" text,
+	"id" uuid PRIMARY KEY NOT NULL,
```
Comment (palisadoes) [inline drizzle_migrations/20250130163044_mature_james_howlett.sql:5]: Why was this file created? There was a migration file there previously. This seems unnecessary.

File: drizzle_migrations/meta/_journal.json [hunk 2]
```diff
@@ -5,8 +5,8 @@
     {
       "idx": 0,
       "version": "7",
-      "when": 1737537615071,
-      "tag": "20250122092015_sweet_scrambler",
+      "when": 1738254644389,
```
Comment (palisadoes) [inline drizzle_migrations/meta/_journal.json:8]: Why was this file created? It seems related to the update previously

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

**Comment by NishantSinghhhhh** [comment]
@palisadoes , I have restored both the files

File: drizzle_migrations/20250130163044_mature_james_howlett.sql (hunk 1 above)
Comment (palisadoes) [reply drizzle_migrations/20250130163044_mature_james_howlett.sql:5]: This is not resolved. The current file in the repo is `20250122092015_sweet_scrambler.sql`
- https://github.com/PalisadoesFoundation/talawa-api/tree/develop-postgres/drizzle_migrations

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. This is not resolved. 
1. The current file in the repo is `20250122092015_sweet_scrambler.sql`
    - https://github.com/PalisadoesFoundation/talawa-api/tree/develop-postgres/drizzle_migrations
1. This does not appear to be a solution
    - `drizzle_migrations/20250130163044_mature_james_howlett.sql`

**Comment by NishantSinghhhhh** [comment]
should I remove [20250130163044_mature_james_howlett.sql] this file ??

**Review by palisadoes — APPROVED** [review]

**Review by palisadoes — COMMENTED** [review]
1. Why is the file added to .gitignore?
    - test/routes/graphql/gql.tada.d.ts
3. Why isn't this one added too?
    - test/routes/graphql/gql.tada-cache.d.ts

**Comment by NishantSinghhhhh** [comment]
> 1. Why is the file added to .gitignore?
>     - test/routes/graphql/gql.tada.d.ts
> 3. Why isn't this one added too?
>     - test/routes/graphql/gql.tada-cache.d.ts

Should I be adding this to git ignore?

**Comment by palisadoes** [comment]
Don't worry. I found out
